### PR TITLE
3897/3879 - Datagrid: Fix table width and horizontal scrollbar inside of modal

### PR DESCRIPTION
**Pull request by ericangeles**
**Explain the _details_ for making this change. What existing problem does the pull request solve?**
It seems that there's a missing class `datagrid-default-modal-width` that will fix the width of the table inside of the modal in #3897. There should be no changes on width being  set to `auto`.

By adding this class, the test page will set the `isInModal` to `true`, where it calculates and stretched the table to fill the datagrid container.

See this line https://github.com/infor-design/enterprise/blob/master/src/components/datagrid/datagrid.js#L4471

In #3879, I've set a width `auto` when the browser viewport is below `565` so horizontal scrollbar will not appear. (Fixes also when resizing it)

**Related github/jira issue (required)**:
Closes
- https://github.com/infor-design/enterprise/issues/3897
- https://github.com/infor-design/enterprise/issues/3879

**Steps necessary to review your pull request (required)**:
- Pull this branch, build, and run the app

1.) #3897

- Navigate to http://localhost:4000/components/datagrid/test-modal-datagrid-search-form.html
- Click `Open Modal` button
- The table should filled the datagrid wrapper
- Toggle the filter button
- It should remain filled the datagrid wrapper
- Test this in all major browsers (Chrome, Firefox, Safari, Edge etc)

2.) #3879 
- Navigate to http://localhost:4000/components/datagrid/test-modal-datagrid-two-columns.html
- Click `Open Modal` button
- Resize the browser up to smaller (mobile) view
- There should be no horizontal scrollbar
- Try it also when the view is already on mobile
- Should work as well

3) Also test any pages on http://localhost:4000/components/datagrid with "modal" mentioned

**Included in this Pull Request**:
- [x] A note to the change log.

### CODE REVIEW

**Comment by tmcconechy** [comment]
Actually this did effect a few test pages in the tests. Maybe have a look. Im also wondering if we can get rid of `datagrid-default-modal-width` with this setting. There are three examples that use that that seem to be failing.

If you can remove that class from those examples and they work still then we can get rid of that as that was a much clunkier solution to this.

**Comment by ericangeles** [comment]
Labeling this as WIP. I will also add the fix in #3879. Seems related to each other.

**Comment by ericangeles** [comment]
Open to review now, @tmcconechy

File: src/components/datagrid/datagrid.js [hunk 1]
```diff
@@ -4470,7 +4470,7 @@ Datagrid.prototype = {
 
       if (!this.elemWidth && this.element.closest('.datagrid-default-modal-width').length > 0) { // handle on invisible modal
         this.elemWidth = this.settings.paging ? 466 : 300; // Default a size for when on modals
-        this.element.css('width', this.elemWidth);
+        this.element.css('width', (!this.isInModal && window.innerWidth > 565) ? this.elemWidth : 'auto');
```
Comment (tmcconechy) [inline src/components/datagrid/datagrid.js:4473]: @ericangeles can you explain the logic on this? Im not sure i follow what its doing? Can you try with 3 columns? 4 ect? I was hoping we could handle this with `-webkit-fill-available` . Or is it actually using that.

The fix works in the steps but just wanted to verify

File: src/components/datagrid/datagrid.js [hunk 2]
```diff
@@ -5963,6 +5963,10 @@ Datagrid.prototype = {
       });
     }
 
+    $(window).on('resize.modal', () => {
```
Comment (tmcconechy) [inline src/components/datagrid/datagrid.js:5966]: Do we need to add something to destroy? And is this needed if the data grid is not on  a modal? Maybe we can put isInModal around it?

File: docs/CHANGELOG.md [hunk 3]
```diff
@@ -15,6 +15,8 @@
 - `[Calendar]` Fixed a bug where the accordion sections would show a line on hover in high contrast mode. ([#2779](https://github.com/infor-design/enterprise/issues/2779))
 - `[Calendar]` Fixed a bug where the days would be out of alignment if the end and starts dates intersect. ([#1725](https://github.com/infor-design/enterprise/issues/1725))
 - `[Lookup]` Fixed an issue in the min width examples that showed up in Safari only. ([#3949](https://github.com/infor-design/enterprise/issues/3949))
+- `[Accordion]` Fixed an issue where the chevron icon is not poperly centered in Safari. ([#2161](https://github.com/infor-design/enterprise/issues/2161))
```
Comment (tmcconechy) [inline docs/CHANGELOG.md:18]: Can you check the alhoabetic order here. Also this accordion entry is duplicated.

**Comment by tmcconechy** [comment]
Any updates here @ericangeles ? Maybe you missed my comments?

**Comment by tmcconechy** [comment]
I added another fix , removing the JS in favor of fully using webkit-fill available. With fallbacks to fixed with on IE and Edge

**Pull request merged**